docs/dwarf-op.rst

```diff
@@ -1,158 +1,180 @@
-                        - See dwf_types table in `dorf_tables` for available types.
-
-``renamejob <name>``: Renames the selected dwarves' custom profession to whatever is specified
-
-**Other Arguments:**
-
-``-help``: displays this help information.
-
-``-debug``: enables debugging print lines
+    dwarf-op --list <table>
+    dwarf-op --reset|--resetall
+    dwarf-op [--select <criteria>] <commands>
+
+Examples
+--------
+
+``dwarf-op --select unoptimized --clear --optimize``
+    Transform newly arrived dwarves into the workers that your fort needs most.
+``dwarf-op --select all --clear --optimize``
+    Rebalance the distribution of skills and attributes across your units.
+``dwarf-op --select [ waves 14 ] --applyjobs [ MINER ]``
+    Make all migrants in wave 14 competent miners.
+``dwarf-op --select [ jobs Stoneworker ] --applytypes [ fast3 strong5 ]``
+    Boost the speed and strength of your masons so they can carry boulders
+    to their workshop faster.
+
+Selection criteria
+------------------
+
+Note that dwarves whose name or profession starts with ``.`` or ``,`` are
+considered "protected", and will not be matched by the selection criteria
+options below unless specifically noted.
+
+You can prepend the letter ``p`` to any option to include protected dwarves in
+your selection. For example, to truly select all dwarves, specify ``pall``
+instead of ``all``.
+
+``highlighted``
+    Selects only the in-game highlighted dwarf (from any screen), regardless of
+    protection status. This is the default if no ``--select`` option is
+    specified.
+``all``
+    Selects all dwarves.
+``<name>``
+    Selects any dwarf with <name> anywhere in their name or nickname. This
+    option ignores protection status.
+``named``
+    Selects dwarves with user-given names.
+``unnamed``
+    Selects dwarves without user-given names.
+``employed```
+    Selects dwarves with custom professions. Does not include optimized dwarves.
+``optimized``
+    Selects dwarves that have been previously optimized by ``dwarf-op``.
+``unoptimized``
+    Selects any dwarves that have not been previously optimized by ``dwarf-op``.
+``protected``
+    Selects protected dwarves.
+``unprotected``
+    Selects unprotected dwarves.
+``drunks``
+    Selects any dwarves who have the ``DRUNK`` profession, including those who
+    have been zeroed by the ``--clear`` command option.
+``jobs "[" jobs <jobname> [<jobname> ...] "]"``
+    Selects any dwarves with the specified custom professions.
+``waves "[" waves <num> [<num> ...] "]"``
+    Selects dwarves from the specified migration waves. Waves are enumerated
+    starting at 0 and increasing by 1 with each wave. The waves go by season and
+    year and thus should match what you see in `list-waves` or Dwarf Therapist.
+    It is recommended that you ``--show`` the selected dwarves before modifying
+    them.
+
+Options
+-------
```

Comment:
If we're grouping all the options together, you missed `--select`

I haven't tested, but I didn't use -<option> for any of the options. So afaik they should all be single dashes. Have you tested this? It doesn't appear that you've modified the arg list.

Reply:
also `--list`. I'll add those in.

As for single vs. double dashes, `dwarf-op` uses `argparse.processArgs` to read the commandline (well, it uses the old name of `utils.processArgs`, but that's just an alias for the argparse version nowadays). That function treats single and double dashes identically. I'm standardizing on getopt-style options (single dash for single letter args, double dash for long name args) so the documented syntax won't change even if the underlying script moves from `argparse.processArgs` to `argparse.processArgsGetopt`, which scripts might do as they become more advanced and need more flexible argument parsing.

Reply:
Oh, very nice